sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -1797,11 +1797,11 @@ SIMPLE_COMMENT
     ;
 
 BRACKETED_EMPTY_COMMENT
-    : '/**/' -> channel(HIDDEN)
+    : '/*' BRACKETED_EMPTY_COMMENT* '*/' -> channel(HIDDEN)
     ;
 
 BRACKETED_COMMENT
-    : '/*' ~[+] .*? '*/' -> channel(HIDDEN)
+    : '/*' (BRACKETED_COMMENT .*? | ~[+] (BRACKETED_COMMENT|.)*?)*? '*/' -> channel(HIDDEN)
```

Comment:
how about
```
: '/*' ((BRACKETED_COMMENT|.)*? | ~[+] (BRACKETED_COMMENT|.)*?) '*/' -> channel(HIDDEN)
```
?

Reply:
I tried it and found that there is a parse conflict with hint.

Reply:
I tried and it works

Reply:
It can't pass `SQLQueryTestSuite`.

Reply:
Ah, I see.
I am not sure about the latest code now. I think it is more reasonable to follow @cloud-fan 's advice.

Reply:
@gengliangwang The lexer rule executed before parser rule. So we must treat it in the lexer rule.